Amplify/Core/Configuration/Internal/AmplifyConfigurationInitialization.swift

```diff
@@ -73,3 +73,74 @@ extension AmplifyConfiguration {
     }
 
 }
+
+
+extension AmplifyOutputsData {
+    init(bundle: Bundle, resource: String) throws {
+        guard let path = bundle.path(forResource: resource, ofType: "json") else {
+            throw ConfigurationError.invalidAmplifyOutputsFile(
+                """
+                Could not load default `\(resource).json` file
+                """,
+
+                """
+                Expected to find the file, `\(resource).json` in the app bundle at `\(bundle.bundlePath)`, but
+                it was not present. Add `\(resource).json` to your app's "Copy Bundle Resources" build
+                phase and invoke `Amplify.configure(with: resource(named: "\(resource)")` with a configuration
+                object that you load. If your resource file is the default `amplify_outputs.json`, you can
+                invoke `Amplify.configure(with: .amplifyOutputs)` instead.
+                """
+            )
+        }
+
+        let url = URL(fileURLWithPath: path)
+
+        self = try AmplifyOutputsData.loadAmplifyConfiguration(from: url)
+    }
+
+    static func loadAmplifyConfiguration(from url: URL) throws -> AmplifyOutputsData {
+        let fileData: Data
+        do {
+            fileData = try Data(contentsOf: url)
+        } catch {
+            throw ConfigurationError.invalidAmplifyOutputsFile(
+                """
+                Could not extract UTF-8 data from `\(url.path)`
+                """,
+
+                """
+                Could not load data from the file at `\(url.path)`. Inspect the file to ensure it is present.
+                The system reported the following error:
+                \(error.localizedDescription)
+                """,
+                error
+            )
+        }
+
+        return try decodeAmplifyConfiguration(from: fileData)
+    }
+
+    static func decodeAmplifyConfiguration(from data: Data) throws -> AmplifyOutputsData {
+        let jsonDecoder = JSONDecoder()
+
+        do {
+            jsonDecoder.keyDecodingStrategy = .convertFromSnakeCase
+            let configuration = try jsonDecoder.decode(AmplifyOutputsData.self, from: data)
+            return configuration
+        } catch {
+            throw ConfigurationError.unableToDecode(
+                """
+                Could not decode `amplify_outputs.json` into a valid AmplifyConfigurationV2 object
```

Comment:
nit: should be - *into a AmplifyOutputsData object*

Reply:
renamed some of the internal method names as well, `decodeAmplifyConfiguration` -> `decodeAmplifyOutputsData`